local-tests/test_catch_up.py

```diff
@@ -26,13 +26,15 @@ def printt(s): print(ctime() + ' | ' + s)
 
 chain.set_flags('no-mdns',
                 port=Seq(30334),
+                validator_port=Seq(30343),
                 ws_port=Seq(9944),
                 rpc_port=Seq(9933),
                 unit_creation_delay=200,
                 execution='Native',
                 pruning='archive')
 addresses = [n.address() for n in chain]
-chain.set_flags(bootnodes=addresses[0], public_addr=addresses)
+validator_addresses = [n.validator_address() for n in chain]
+chain.set_flags(bootnodes=addresses[0], public_addr=addresses, public_validator_addresses=validator_addresses)
```

Comment:
nit: if this is smt that only validators use consider using `chain.set_flags_validator` to only set it for validator nodes

Reply:
Changed, although hm, now `public_addr` is also only present in validators, I hope this does not break anything...

This also required some additional changes in accepting the lack of this parameter and panicking when it's not present on a validator.